treatment-service/models/storage.go

```diff
@@ -201,13 +202,14 @@ func (i *ExperimentIndex) matchRealSetSegment(segmentName string, value float64)
 }
 
 func (i *ExperimentIndex) matchSegment(segmentName string, values []*_segmenters.SegmenterValue) Match {
-	_, exists := i.Experiment.Segments[segmentName]
-	if !exists {
-		// Optional segmenter, first element from segmenter values (cannot be empty)
-		return Match{Strength: MatchStrengthWeak, Value: values[0]}
+	if len(values) == 0 {
```

Comment:
We only need to have this check for optional segmenter here for `len(values) == 0` - the rest is handled within the check for each type of value (eg: `matchFlagSetSegment` or `matchStringSetSegment`)